File: bindings/javascript/Polycode/LocalShaderParam.js
function LocalShaderParam() {
	if(arguments[0] != "__skip_ptr__") {
		this.__ptr = Polycode.LocalShaderParam()
	}
	Object.defineProperties(this, {
		'name': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_name, set: LocalShaderParam.prototype.__set_name},
		'type': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_type, set: LocalShaderParam.prototype.__set_type},
		'ownsPointer': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_ownsPointer, set: LocalShaderParam.prototype.__set_ownsPointer},
		'arraySize': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_arraySize, set: LocalShaderParam.prototype.__set_arraySize},
		'param': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_param, set: LocalShaderParam.prototype.__set_param},
		'accessMutex': { enumerable: true, configurable: true, get: LocalShaderParam.prototype.__get_accessMutex, set: LocalShaderParam.prototype.__set_accessMutex}
	})
}

LocalShaderParam.prototype.__get_name = function() {
	return Polycode.LocalShaderParam__get_name(this.__ptr)
}

LocalShaderParam.prototype.__set_name = function(val) {
	Polycode.LocalShaderParam__set_name(this.__ptr, val)
}

LocalShaderParam.prototype.__get_type = function() {
	return Polycode.LocalShaderParam__get_type(this.__ptr)
}

LocalShaderParam.prototype.__set_type = function(val) {
	Polycode.LocalShaderParam__set_type(this.__ptr, val)
}

LocalShaderParam.prototype.__get_ownsPointer = function() {
	return Polycode.LocalShaderParam__get_ownsPointer(this.__ptr)
}

LocalShaderParam.prototype.__set_ownsPointer = function(val) {
	Polycode.LocalShaderParam__set_ownsPointer(this.__ptr, val)
}

LocalShaderParam.prototype.__get_arraySize = function() {
	return Polycode.LocalShaderParam__get_arraySize(this.__ptr)
}

LocalShaderParam.prototype.__set_arraySize = function(val) {
	Polycode.LocalShaderParam__set_arraySize(this.__ptr, val)
}

LocalShaderParam.prototype.__get_param = function() {
	var retVal = new ProgramParam()
	retVal.__ptr = 	Polycode.LocalShaderParam__get_param(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.__set_param = function(val) {
	Polycode.LocalShaderParam__set_param(this.__ptr, val.__ptr)
}

LocalShaderParam.prototype.__get_accessMutex = function() {
	var retVal = new CoreMutex()
	retVal.__ptr = 	Polycode.LocalShaderParam__get_accessMutex(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.__set_accessMutex = function(val) {
	Polycode.LocalShaderParam__set_accessMutex(this.__ptr, val.__ptr)
}

Duktape.fin(LocalShaderParam.prototype, function (x) {
	if (x === LocalShaderParam.prototype) {
		return;
	}
	Polycode.LocalShaderParam__delete(x.__ptr)
})

LocalShaderParam.prototype.Copy = function() {
	var retVal = new LocalShaderParam()
	retVal.__ptr = Polycode.LocalShaderParam_Copy(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.getNumber = function() {
	return Polycode.LocalShaderParam_getNumber(this.__ptr)
}

LocalShaderParam.prototype.getVector2 = function() {
	var retVal = new Vector2()
	retVal.__ptr = Polycode.LocalShaderParam_getVector2(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.getVector3 = function() {
	var retVal = new Vector3()
	retVal.__ptr = Polycode.LocalShaderParam_getVector3(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.getMatrix4 = function() {
	var retVal = new Matrix4()
	retVal.__ptr = Polycode.LocalShaderParam_getMatrix4(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.getColor = function() {
	var retVal = new Color()
	retVal.__ptr = Polycode.LocalShaderParam_getColor(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.setNumber = function(x) {
	Polycode.LocalShaderParam_setNumber(this.__ptr, x)
}

LocalShaderParam.prototype.setVector2 = function(x) {
	Polycode.LocalShaderParam_setVector2(this.__ptr, x)
}

LocalShaderParam.prototype.setVector3 = function(x) {
	Polycode.LocalShaderParam_setVector3(this.__ptr, x)
}

LocalShaderParam.prototype.setMatrix4 = function(x) {
	Polycode.LocalShaderParam_setMatrix4(this.__ptr, x)
}

LocalShaderParam.prototype.setMatrix4Array = function(x) {
	Polycode.LocalShaderParam_setMatrix4Array(this.__ptr, x)
}

LocalShaderParam.prototype.setColor = function(x) {
	Polycode.LocalShaderParam_setColor(this.__ptr, x)
}

LocalShaderParam.prototype.setTexture = function(texture) {
	Polycode.LocalShaderParam_setTexture(this.__ptr, texture)
}

LocalShaderParam.prototype.getTexture = function() {
	var retVal = new Texture()
	retVal.__ptr = Polycode.LocalShaderParam_getTexture(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.setCubemap = function(cubemap) {
	Polycode.LocalShaderParam_setCubemap(this.__ptr, cubemap)
}

LocalShaderParam.prototype.getCubemap = function() {
	var retVal = new Cubemap()
	retVal.__ptr = Polycode.LocalShaderParam_getCubemap(this.__ptr)
	return retVal
}

LocalShaderParam.prototype.setParamValueFromString = function(type,pvalue) {
	Polycode.LocalShaderParam_setParamValueFromString(this.__ptr, type, pvalue)
}

File: bindings/lua/Polycode/LocalShaderParam.lua
class "LocalShaderParam"


function LocalShaderParam:__getvar(name)
	if name == "name" then
		return Polycode.LocalShaderParam_get_name(self.__ptr)
	elseif name == "type" then
		return Polycode.LocalShaderParam_get_type(self.__ptr)
	elseif name == "ownsPointer" then
		return Polycode.LocalShaderParam_get_ownsPointer(self.__ptr)
	elseif name == "arraySize" then
		return Polycode.LocalShaderParam_get_arraySize(self.__ptr)
	end
end

function LocalShaderParam:__setvar(name,value)
	if name == "name" then
		Polycode.LocalShaderParam_set_name(self.__ptr, value)
		return true
	elseif name == "type" then
		Polycode.LocalShaderParam_set_type(self.__ptr, value)
		return true
	elseif name == "ownsPointer" then
		Polycode.LocalShaderParam_set_ownsPointer(self.__ptr, value)
		return true
	elseif name == "arraySize" then
		Polycode.LocalShaderParam_set_arraySize(self.__ptr, value)
		return true
	end
	return false
end
function LocalShaderParam:LocalShaderParam(...)
	local arg = {...}
	for k,v in pairs(arg) do
		if type(v) == "table" then
			if v.__ptr ~= nil then
				arg[k] = v.__ptr
			end
		end
	end
	if self.__ptr == nil and arg[1] ~= "__skip_ptr__" then
		self.__ptr = Polycode.LocalShaderParam(unpack(arg))
	end
end

function LocalShaderParam:Copy()
	local retVal =  Polycode.LocalShaderParam_Copy(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<LocalShaderParam>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:getNumber()
	local retVal =  Polycode.LocalShaderParam_getNumber(self.__ptr)
	return retVal
end

function LocalShaderParam:getVector2()
	local retVal =  Polycode.LocalShaderParam_getVector2(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector2"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:getVector3()
	local retVal =  Polycode.LocalShaderParam_getVector3(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Vector3"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:getMatrix4()
	local retVal =  Polycode.LocalShaderParam_getMatrix4(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Matrix4"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:getColor()
	local retVal =  Polycode.LocalShaderParam_getColor(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["Color"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:setNumber(x)
	local retVal = Polycode.LocalShaderParam_setNumber(self.__ptr, x)
end

function LocalShaderParam:setVector2(x)
	local retVal = Polycode.LocalShaderParam_setVector2(self.__ptr, x.__ptr)
end

function LocalShaderParam:setVector3(x)
	local retVal = Polycode.LocalShaderParam_setVector3(self.__ptr, x.__ptr)
end

function LocalShaderParam:setMatrix4(x)
	local retVal = Polycode.LocalShaderParam_setMatrix4(self.__ptr, x.__ptr)
end

function LocalShaderParam:setMatrix4Array(x)
	local retVal = Polycode.LocalShaderParam_setMatrix4Array(self.__ptr, x.__ptr)
end

function LocalShaderParam:setColor(x)
	local retVal = Polycode.LocalShaderParam_setColor(self.__ptr, x.__ptr)
end

function LocalShaderParam:setTexture(texture)
	local retVal = Polycode.LocalShaderParam_setTexture(self.__ptr, texture.__ptr)
end

function LocalShaderParam:getTexture()
	local retVal =  Polycode.LocalShaderParam_getTexture(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<Texture>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:setCubemap(cubemap)
	local retVal = Polycode.LocalShaderParam_setCubemap(self.__ptr, cubemap.__ptr)
end

function LocalShaderParam:getCubemap()
	local retVal =  Polycode.LocalShaderParam_getCubemap(self.__ptr)
	if retVal == nil then return nil end
	local __c = _G["shared_ptr<Cubemap>"]("__skip_ptr__")
	__c.__ptr = retVal
	return __c
end

function LocalShaderParam:__delete()
	if self then Polycode.delete_LocalShaderParam(self.__ptr) end
end
